CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+# Minimum version follows the current Ubuntu LTS and RHEL version
+cmake_minimum_required(VERSION 3.20)
+# CMake version compatibility. This allows to future-proof implementations and easily update
+# when bumping cmake_minimum_required
+# Enable using return(PROPAGATE)
+# TODO: Remove when cmake 3.25 is commonly distributed
+if (POLICY CMP0140)
+	cmake_policy(SET CMP0140 NEW)
+endif ()
+
+#[==============================================================================================[
+#                                   Basic project definition                                   #
+]==============================================================================================]
+
+list(APPEND CMAKE_MESSAGE_CONTEXT Template)
+project(Template
+		VERSION 0.1.0
+		DESCRIPTION "CMake project template"
+		HOMEPAGE_URL https://github.com/LecrisUT/CMake-Template
+		LANGUAGES CXX
+)
+
+# Back-porting to PROJECT_IS_TOP_LEVEL to older cmake
+# TODO: Remove when requiring cmake >= 3.21
+if (NOT DEFINED Template_IS_TOP_LEVEL)
+	if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+		set(PROJECT_IS_TOP_LEVEL ON)
+	else ()
+		set(PROJECT_IS_TOP_LEVEL OFF)
+	endif ()
+	set(Template_IS_TOP_LEVEL ${PROJECT_IS_TOP_LEVEL})
+endif ()
+
+# Specify C++ standard used in the project.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+#[==============================================================================================[
+#                                            Options                                            #
+]==============================================================================================]
```

Comment:
Simplest point is the usage of some "banner" comments to highlight sections where the user might be interested in looking for what options are available